internal/fields/dependency_manager.go

```diff
@@ -145,8 +145,12 @@ func (dm *DependencyManager) injectFieldsWithRoot(root string, defs []common.Map
 			}
 
 			transformed := transformImportedField(imported)
```

Comment:
I'm wondering if it isn't possible to sketch the implementation without involving the `DeepUpdate` method (less error-prone code).

For example:

```
base := def
base.Delete("external")
transformed := transformImportedField(base, imported)
```

Function `transformImportedField` can create new `base` struct if it's nil.

Reply:
I will give a try, but this code would modify the definition, and I wanted to avoid this. For that we need a way to clone it and use as the base struct, or to deep update the final map.
Would you prefer clone over deep update?

Reply:
I just wouldn't like to add more potentially error-prone methods to `mapStr`, but if you feel confident the function maybe it's worth leaving it as is.

Reply:
Well, I copied it from libbeat where it has served well for years :slightly_smiling_face:

Maybe in the future we can refactor these methods to use `FieldDefinition`s instead of `mapStr`s.

Reply:
I remember that Stefen strongly recommended not to use `mapStr` anymore :)

Anyway, I see that you covered this part with tests, so it shouldn't be a big unknown. Let's not elongate the discussion.